src/utils/calculator.js

```diff
@@ -0,0 +1,132 @@
+const MAXIMUM = 999;
+const ERROR = {
+  maximum: "숫자는 세 자리까지만 입력 가능합니다!",
+  operator: "숫자를 먼저 입력한 후 연산자를 입력해주세요!",
+  maxOperator: "두개의 숫자 연산만 가능합니다!",
+};
+export const INITIAL = "0";
+
+const INFINITY = "오류";
+
+const OPERATOR = ["＋", "-", "X", "/"];
+const OPERATOR_REGEX = new RegExp(
+  `(${OPERATOR.map((op) => `\\${op}`).join("|")})`
+);
+
+const CALCULATE = "=";
+
+class Calculator {
+  operators = {
+    "＋": this.sum,
+    "-": this.sub,
+    X: this.mul,
+    "/": this.div,
+  };
+
+  isInfinity(result) {
+    return result === Infinity ? INFINITY : result;
+  }
+
+  update(value, total) {
+    if (total === INFINITY) {
+      return total;
+    } else if (OPERATOR.includes(value)) {
+      return this.updateOperator(value, total);
+    } else if (value === CALCULATE) {
+      return this.updateCalculate(total);
+    } else {
+      return this.updateNumber(value, total);
+    }
+  }
+
+  updateNumber(value, total) {
+    const parts = total.split(OPERATOR_REGEX);
+    const lastPart = parts[parts.length - 1];
+
+    const updateValue = Number(lastPart + "" + value);
+
+    if (updateValue > MAXIMUM) {
+      window.alert(ERROR.maximum);
+      return total;
+    }
+
+    parts[parts.length - 1] = updateValue.toString();
+
+    return parts.join("");
+  }
+
+  checkOperator(lastChar, total) {
+    return OPERATOR.includes(lastChar) || total === INITIAL;
+  }
+
+  checkMaxOperator(total) {
+    let check = false;
+
+    OPERATOR.forEach((oper) => {
+      if (total.includes(oper)) {
+        check = true;
+      }
+    });
+
+    return check;
```

Comment:
억제 가능한 지역 변수는 최대한 지양하는 게 코드 읽을 때 큰 도움이 되어서...
아래처럼 해도 동일한 기능을 할 것 같군요!

```suggestion
    return OPERATOR.some(total.includes);
```